### PR TITLE
postgresql-*: move oci entrypoint to /usr/local/bin

### LINKED ISSUE #9903: postgresql-x-oci-entrypoint: Move entrypoint script out of /var/lib/postgres/initdb
👋🏻 Hello friends!

The postgres image is giving me a bit of trouble because the [container entrypoint lives in `/var/lib/postgres/initdb`](https://github.com/wolfi-dev/os/blob/7119343d977d22c24b3c320ff07eae126613ba9d/postgresql-13.yaml#L101-L103) but then also [points back at that same directory](https://github.com/wolfi-dev/os/blob/7119343d977d22c24b3c320ff07eae126613ba9d/postgresql/postgresql-entrypoint.sh#L323) to load init scripts (be they sql or shell or whatever). This is different from Docker library postgres image in which the entrypoint lives in `/usr/local/bin` and the init scripts are loaded from `/docker-entrypoint-initdb.d/`

The wolfi situation presents two issues
- Since the script automatically loads shells scripts (at least based on .sh suffix) its a little surprising that it doesn't load itself automatically and just create some wierd inifinite loop 😱  That doesn't happen in practice, but it feels like its possible to me. 🤔 
- The second is the point of that directory (`/var/lib/postgres/initdb` in the wolfi case) is to load up sql scripts or what ever from a config map or other volume and the fact that the entrypoint is in that same directory means you need to do so carefully so that you don't overwrite the entrypoint and prevent your container from starting

Lets move that entrypoint over to /usr/local/bin or something?

### PR DESCRIPTION
**Pull request by nsmith5**
Fixes: #9903

I've left a symlink back to the original path hopefully make the change not breaking. Not sure if thats the right approach though! Limme know ❤️

### CODE REVIEW

File: postgresql-12.yaml [hunk 1]
```diff
@@ -100,8 +100,9 @@ subpackages:
     pipeline:
       - runs: |
           mkdir -p ${{targets.subpkgdir}}/var/lib/postgres/initdb
-          cp postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/
-          chmod +x ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
+          cp postgresql-entrypoint.sh ${{targets.subpkgdir}}/usr/local/bin
+          chmod +x ${{targets.subpkgdir}}/usr/local/bin/postgresql-entrypoint.sh
+          ln -s ${{targets.subpkgdir}}/usr/local/bin/postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
```
Comment (joshrwolf) [inline postgresql-12.yaml:105]:
```suggestion
          install -Dm755 postgresql-entrypoint.sh ${{targets.contextdir}}/usr/bin/postgresql-entrypoint.sh
          ln -s ${{targets.subpkgdir}}/usr/bin/postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
```

File: postgresql-13.yaml [hunk 2]
```diff
@@ -99,8 +99,9 @@ subpackages:
     pipeline:
       - runs: |
           mkdir -p ${{targets.subpkgdir}}/var/lib/postgres/initdb
-          cp postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/
-          chmod +x ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
+          cp postgresql-entrypoint.sh ${{targets.subpkgdir}}/usr/local/bin
+          chmod +x ${{targets.subpkgdir}}/usr/local/bin/postgresql-entrypoint.sh
+          ln -s ${{targets.subpkgdir}}/usr/local/bin/postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
```
Comment (joshrwolf) [inline postgresql-13.yaml:104]:
```suggestion
          install -Dm755 postgresql-entrypoint.sh ${{targets.contextdir}}/usr/bin/postgresql-entrypoint.sh
          ln -s ${{targets.subpkgdir}}/usr/local/bin/postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
```

File: postgresql-14.yaml [hunk 3]
```diff
@@ -100,8 +100,9 @@ subpackages:
     pipeline:
       - runs: |
           mkdir -p ${{targets.subpkgdir}}/var/lib/postgres/initdb
-          cp postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/
-          chmod +x ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
+          cp postgresql-entrypoint.sh ${{targets.subpkgdir}}/usr/local/bin
+          chmod +x ${{targets.subpkgdir}}/usr/local/bin/postgresql-entrypoint.sh
+          ln -s ${{targets.subpkgdir}}/usr/local/bin/postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
```
Comment (joshrwolf) [inline postgresql-14.yaml:105]:
```suggestion
          install -Dm755 postgresql-entrypoint.sh ${{targets.contextdir}}/usr/bin/postgresql-entrypoint.sh
          ln -s ${{targets.subpkgdir}}/usr/local/bin/postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
```

File: postgresql-15.yaml [hunk 4]
```diff
@@ -96,8 +96,9 @@ subpackages:
     pipeline:
       - runs: |
           mkdir -p ${{targets.subpkgdir}}/var/lib/postgres/initdb
-          cp postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/
-          chmod +x ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
+          cp postgresql-entrypoint.sh ${{targets.subpkgdir}}/usr/local/bin
+          chmod +x ${{targets.subpkgdir}}/usr/local/bin/postgresql-entrypoint.sh
+          ln -s ${{targets.subpkgdir}}/usr/local/bin/postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
```
Comment (joshrwolf) [inline postgresql-15.yaml:101]:
```suggestion
          install -Dm755 postgresql-entrypoint.sh ${{targets.contextdir}}/usr/bin/postgresql-entrypoint.sh
          ln -s ${{targets.subpkgdir}}/usr/local/bin/postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
```

File: postgresql-16.yaml [hunk 5]
```diff
@@ -97,8 +97,9 @@ subpackages:
     pipeline:
       - runs: |
           mkdir -p ${{targets.subpkgdir}}/var/lib/postgres/initdb
-          cp postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/
-          chmod +x ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
+          cp postgresql-entrypoint.sh ${{targets.subpkgdir}}/usr/local/bin
+          chmod +x ${{targets.subpkgdir}}/usr/local/bin/postgresql-entrypoint.sh
+          ln -s ${{targets.subpkgdir}}/usr/local/bin/postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
```
Comment (joshrwolf) [inline postgresql-16.yaml:102]:
```suggestion
          install -Dm755 postgresql-entrypoint.sh ${{targets.contextdir}}/usr/bin/postgresql-entrypoint.sh
          ln -s ${{targets.subpkgdir}}/usr/local/bin/postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
```

**Comment by joshrwolf** [comment]
oh hey stranger 👋 

the approach overall lgtm, thank you!

just tweaked the method a bit and put things in `/usr/bin` to align with the wolfi standards

File: postgresql-13.yaml [hunk 6]
```diff
@@ -99,8 +99,8 @@ subpackages:
     pipeline:
       - runs: |
           mkdir -p ${{targets.subpkgdir}}/var/lib/postgres/initdb
-          cp postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/
-          chmod +x ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
+          install -Dm755 postgresql-entrypoint.sh ${{targets.contextdir}}/usr/bin/postgresql-entrypoint.sh
+          ln -s ${{targets.subpkgdir}}/usr/local/bin/postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
```
Comment (nsmith5) [inline postgresql-13.yaml:103]:
```suggestion
          ln -s ${{targets.subpkgdir}}/usr/bin/postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
```

File: postgresql-14.yaml [hunk 7]
```diff
@@ -100,8 +100,8 @@ subpackages:
     pipeline:
       - runs: |
           mkdir -p ${{targets.subpkgdir}}/var/lib/postgres/initdb
-          cp postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/
-          chmod +x ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
+          install -Dm755 postgresql-entrypoint.sh ${{targets.contextdir}}/usr/bin/postgresql-entrypoint.sh
+          ln -s ${{targets.subpkgdir}}/usr/local/bin/postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
```
Comment (nsmith5) [inline postgresql-14.yaml:104]:
```suggestion
          ln -s ${{targets.subpkgdir}}/usr/bin/postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
```

File: postgresql-15.yaml [hunk 8]
```diff
@@ -96,8 +96,8 @@ subpackages:
     pipeline:
       - runs: |
           mkdir -p ${{targets.subpkgdir}}/var/lib/postgres/initdb
-          cp postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/
-          chmod +x ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
+          install -Dm755 postgresql-entrypoint.sh ${{targets.contextdir}}/usr/bin/postgresql-entrypoint.sh
+          ln -s ${{targets.subpkgdir}}/usr/local/bin/postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
```
Comment (nsmith5) [inline postgresql-15.yaml:100]:
```suggestion
          ln -s ${{targets.subpkgdir}}/usr/bin/postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
```

File: postgresql-16.yaml [hunk 9]
```diff
@@ -97,8 +97,8 @@ subpackages:
     pipeline:
       - runs: |
           mkdir -p ${{targets.subpkgdir}}/var/lib/postgres/initdb
-          cp postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/
-          chmod +x ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
+          install -Dm755 postgresql-entrypoint.sh ${{targets.contextdir}}/usr/bin/postgresql-entrypoint.sh
+          ln -s ${{targets.subpkgdir}}/usr/local/bin/postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
```
Comment (nsmith5) [inline postgresql-16.yaml:101]:
```suggestion
          ln -s ${{targets.subpkgdir}}/usr/bin/postgresql-entrypoint.sh ${{targets.subpkgdir}}/var/lib/postgres/initdb/postgresql-entrypoint.sh
```

**Review by nsmith5 — COMMENTED** [review]
Sweet! Just need to update a couple more of the symlinks to /usr/bin instead of /usr/local/bin now I think

**Comment by joshrwolf** [comment]
> Sweet! Just need to update a couple more of the symlinks to /usr/bin instead of /usr/local/bin now I think

gah sorry, suggestion-committing is hard.

lgtm, thanks :nathan-excite:

**Review by joshrwolf — APPROVED** [review]

**Pull request merged**